src/platform/input/sdl2mouse.cpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * @file
+ *
+ * @author feliwir
+ *
+ * @brief SDL2 Mouse Input interface.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
+#include "sdl2mouse.h"
+#include "editmain.h"
+#include "gameclient.h"
+#include <cstdio>
+
+SDL_Cursor *SDL2Mouse::s_loadedCursors[CURSOR_COUNT][8];
+
+SDL2Mouse::SDL2Mouse() : m_nextFreeIndex(0), m_nextGetIndex(0) {}
+
+void SDL2Mouse::Init()
+{
+    Mouse::Init();
+
+    std::memset(m_eventBuffer, 0, sizeof(m_eventBuffer));
+    std::memset(s_loadedCursors, 0, sizeof(s_loadedCursors));
+}
+
+void SDL2Mouse::Load_Cursor(const char *file_location, uint32_t load_index, uint32_t load_direction) const
+{
+    if (s_loadedCursors[load_index][load_direction] != NULL) {
+        // Cursor at this position already loaded. Ignoring subsequent load.
+        return;
+    }
+    // TODO: this should parse the ANI file and load the cursor correctly
+    SDL_Surface *surface = SDL_LoadBMP(file_location);
+    if (!surface) {
+        // Failed to load surface
+        return;
+    }
+    SDL_Cursor *cursor = SDL_CreateColorCursor(surface, 0, 0);
+    if (!cursor) {
+        // Failed to create cursor
+        return;
```

Comment:
Since there is no error handling for this function, I suggest to at least log some errors.